### PR TITLE
rds_instance - fix idempotency when using non-lowercase for preferred_maintenance_window

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Rather than relying on changing the docs (#310) force the day of the week into lowercase so that it matches what AWS returns.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

rds_instance

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by tremble** [comment]
note: tests fail without #515

File: changelogs/516-rds_instance-preferred_maintenance_window.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- rds_instance - convert ``preferred_maintenance_window`` days into lowercase so changed returns properly (https://github.com/ansible-collections/community.aws/pull/515).
```
Comment (markuman) [inline changelogs/516-rds_instance-preferred_maintenance_window.yml:2]: Url shows pr 515, but this is 516

**Review by markuman — CHANGES_REQUESTED** [review]
only changelog needs link the the correct pr. otherwise LGTM

File: changelogs/516-rds_instance-preferred_maintenance_window.yml (hunk 1 above)
Comment (tremble) [inline changelogs/516-rds_instance-preferred_maintenance_window.yml:2]:
```suggestion
- rds_instance - convert ``preferred_maintenance_window`` days into lowercase so changed returns properly (https://github.com/ansible-collections/community.aws/pull/516).
```

**Comment by tremble** [comment]
Tests also need #531 for the full suite to run.

File: changelogs/516-rds_instance-preferred_maintenance_window.yml (hunk 1 above)
Comment (tremble) [inline changelogs/516-rds_instance-preferred_maintenance_window.yml:2]:
```suggestion
- rds_instance - convert ``preferred_maintenance_window`` days into lowercase so changed returns properly (https://github.com/ansible-collections/community.aws/pull/516).
```

**Review by alinabuzachis — APPROVED** [review]
LGTM.

**Comment by tremble** [comment]
Thanks for the reviews @markuman @alinabuzachis

**Pull request merged**